### ISSUE TITLE
NllbTokenizer

### ISSUE THREAD

**Issue by beginner-byte** [huggingface/swift-transformers#107]
.unsupportedTokenizer("NllbTokenizer")

**Comment by fumoboy007** [comment]
+1 Looking forward to using a CoreML implementation of the [NLLB-200](https://ai.meta.com/blog/nllb-200-high-quality-machine-translation/) model for translations in my Swift app!

I guess we would need to port [`NllbTokenizerFast`](https://github.com/huggingface/transformers/blob/main/src/transformers/models/nllb/tokenization_nllb_fast.py) to Swift? Maybe I can help with that. @pcuenca How difficult would it be for a n00b to do this?

**Comment by pcuenca** [comment]
Hello! Sorry it took so long to get to this.

The best way for tokenizers to become compatible with swift-transformers is to make them compatible with the tokenizers library, so they can be automatically downloaded from configuration and vocabulary files in the Hub. This is still not the case for the `NllbTokenizer`, which uses a custom implementation in transformers, as you saw. However, @xenova already did all the legwork required to convert this family of tokenizers. Here's [the code](https://github.com/huggingface/transformers/pull/25478) he used for conversion, and here's one of the tokenizers he converted: https://huggingface.co/Xenova/nllb-200-distilled-600M.

In summary, tokenizer [Xenova/nllb-200-distilled-600M](https://huggingface.co/Xenova/nllb-200-distilled-600M) is equivalent to [facebook/nllb-200-distilled-600M](https://huggingface.co/facebook/nllb-200-distilled-600M/blob/main/tokenizer_config.json), but it can be loaded by swift-transformers (and transformers.js).

This [simple test case we just added](https://github.com/huggingface/swift-transformers/blob/5059cd47c52952f4f70e4f1bfbc907f0879c2b36/Tests/TokenizersTests/TokenizerTests.swift#L124) shows that the tokenizer can now be loaded and produces outputs equivalent to the Python version.

Please, let us know if you face any issues using this tokenizer.

**Comment by FL33TW00D** [comment]
If there's nothing else, we can close this! Please ping again with any more issues @fumoboy007 @beginner-byte
